Patch to include the Oberbank

statementaustrianoberbank opened the pull request:
Warning: This is my first commit, and I am not fluent in Python. I have mainly modified the existing raiffeisen plugin to make it read my Oberbank file.

Problems:
I could not figure out how to tell ofxstatement to delete the first line (which contains the colum names). So if you do not delete the first line with the colum names, you will get an error, similar to the one attached.

I have not modified the test case, so one for Oberbank is still missing.
[error.txt](https://github.com/nblock/ofxstatement-austrian/files/850171/error.txt)

Thanks in advance for looking at this!

statementaustrianoberbank committed:
Sample file for Oberbank

With the same values in it as in the raiffeisen one, so the test can be easier adapted.

statementaustrianoberbank committed:
Added the plugin for oberbank

this is just a modified version of the raiffeisen one

statementaustrianoberbank committed:
Updated setup file to include the oberbank plugin

Not sure if this is the correct way to do this

statementaustrianoberbank committed:
Original file from the Oberbank web interface

just anonymized

statementaustrianoberbank committed:
Added the oberbank to the description

statementaustrianoberbank committed:
Fixed line 22 so the travis build does not fail

nblock commented on src/ofxstatement/plugins/oberbank.py:
```diff
@@ -0,0 +1,65 @@
+
+    def parse_record(self, line):
+        """Parse a single record."""
+        # Currency
```
You could use the following snippet to skip the first line:
```
# Skip header line
if self.cur_record == 1:
    return None
```

nblock reviewed (changes requested):
It would be great to use this file as 'oberbank.csv' sample instead of the sample you provided. Just to avoid copy-paste errors.

However, it contains private data such as IBANs. Would you mind to anonymize those entries and commit this file as 'oberbank.csv'?

nblock commented:
Thank you very much for your contribution! Once you've updated your commits, tests can be added for the Oberbank plugin.

statementaustrianoberbank committed:
Ignore the first line in the csv file

As this will lead to errors since the first line contains names of the data instead of data itself.

statementaustrianoberbank committed:
Use the original file from the web interface

Now using the original (yet anonymized) file that I have downloaded from the Oberbank web interface. The IBAN and BIC are samples I have found on the web. They do not fit together (IBAN belongs to an austrian account, yet BIC is from a german Bank).

statementaustrianoberbank committed:
removed because it is now called oberbank.csv

statementaustrianoberbank commented:
I have added the code you suggested to skip the first line and changed the filename for the original file to oberbank.csv. It was already anonymized, so I am not sure if you want me to completely remove the IBAN (it is not mine) or if it can stay.

nblock commented:
LGTM, thx for the improvements!
Are you going to provide a test for the plugin or should I take it from there?

statementaustrianoberbank commented:
It would be great if you can take it from here, as I have never written a test before and know nothing about the whole process. If you do not have time for this, then I could try to adapt the others so that they fit. Thanks again!

nblock commented:
OK, I'll take it from here.

nblock commented:
Fixed with 8314e5569bfd673893f71a4fc3f55097ce4ee6fc.
